tilelang/intrinsics/mfma_macro_generator.py

```
@@ -33,6 +34,20 @@
 
+def _is_gfx950() -> bool:
+    """Detect whether the current device is GFX950 via torch."""
+    try:
+        import torch
+
+        if torch.version.hip is None or not torch.cuda.is_available():
+            return False
+        props = torch.cuda.get_device_properties(0)
+        gcn_arch = getattr(props, "gcnArchName", "")
+        return gcn_arch.startswith("gfx950")
+    except Exception:
```

Ruff (0.15.2)

[warning] Do not catch blind exception: `Exception`

(BLE001)